File: backend/modules/crm/schemas/enrichment.py
"""Enrichment schemas — shape que o frontend consome."""

from typing import Any

from pydantic import BaseModel, Field


class CNPJEnrichment(BaseModel):
    cnpj: str
    razao_social: str | None = None
    nome_fantasia: str | None = None
    cnae_principal: str | None = Field(
        default=None,
        description="Código + descrição do CNAE principal",
    )
    cnaes_secundarios: list[str] = []
    qsa: list[dict[str, Any]] = []
    capital_social: float | None = None
    situacao: str | None = None
    endereco: dict[str, str | None] = {}
    telefone: str | None = None
    porte: str | None = None
    data_abertura: str | None = None
    simples_nacional: bool | None = None
    cache_hit: bool = False


class CEPEnrichment(BaseModel):
    cep: str
    logradouro: str | None = None
    bairro: str | None = None
    cidade: str | None = None
    uf: str | None = None
    coordenadas: dict[str, Any] | None = None
    cache_hit: bool = False


class TaxaItem(BaseModel):
    nome: str
    valor: float


class TaxasResponse(BaseModel):
    taxas: list[TaxaItem] = []
    selic: float | None = None
    cdi: float | None = None
    ipca: float | None = None
    cache_hit: bool = False

File: frontend/src/types/crm/enrichment.ts
export interface CNPJEnrichment {
  cnpj: string;
  razao_social: string | null;
  nome_fantasia: string | null;
  cnae_principal: string | null;
  cnaes_secundarios: string[];
  qsa: Array<{
    nome_socio: string | null;
    qualificacao_socio: string | null;
    data_entrada_sociedade: string | null;
  }>;
  capital_social: number | null;
  situacao: string | null;
  endereco: {
    logradouro: string | null;
    numero: string | null;
    complemento: string | null;
    bairro: string | null;
    municipio: string | null;
    uf: string | null;
    cep: string | null;
  };
  telefone: string | null;
  porte: string | null;
  data_abertura: string | null;
  simples_nacional: boolean | null;
  cache_hit: boolean;
}

export interface CEPEnrichment {
  cep: string;
  logradouro: string | null;
  bairro: string | null;
  cidade: string | null;
  uf: string | null;
  coordenadas: { latitude: string; longitude: string } | null;
  cache_hit: boolean;
}

export interface TaxasResponse {
  taxas: Array<{ nome: string; valor: number }>;
  selic: number | null;
  cdi: number | null;
  ipca: number | null;
  cache_hit: boolean;
}
